WordPress/Classes/Extensions/AbstractPost+fixedMediaLocalURL.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+
+extension AbstractPost {
+    /// If a post is in the failed state it can contain references to local files.
+    /// When updating the app through the App Store the local paths can change.
+    /// This will fix any outdated local path with the correct one.
+    ///
+    func fixedMediaLocalURL() {
+        guard isFailed,
+            var content = self.content else {
+            return
+        }
+
+        media.forEach { media in
+            guard !media.hasRemote else {
+                return
+            }
+
+            extractFilenameAndAbsolutePath(from: media)?.forEach { filename, absolutePath in
+                content = content.replacingMatches(of: "src\\s*=\\s*\"(file://.[^\"]+?/\(NSRegularExpression.escapedPattern(for: filename)))\"", with: "src=\"\(absolutePath)\"")
```

Comment:
I'm suuuuuper worried about the use of regexes here. I tried reading this and I _think_ it's correct, but I still don't like it.

There's some precedent in the app on how to handle this, the `PostCoordinator` does something really similar for when a `Media` objects gets correctly uploaded, and it's piggy-backing on the `HTMLProcessor` from `Aztec`: https://github.com/wordpress-mobile/WordPress-iOS/blob/b71fc07ee9a9345ec24dbaa08b931a618402f021/WordPress/Classes/Services/PostCoordinator.swift#L184-L213

It's much more verbose than your option, but it's been extensively used elsewhere already, making me much confident in it being reliable.

Reply:
Those seem like good reason to do it the way it's currently done then :)

So, @leandroalonso, I think I'd love for two things to happen:
a) as @diegoreymendez mentioned, splitting out the regex to a separate property/method with explanation of what it does
b) I'd also love to see a comment explaining the trade-off you mentioned above directly in the code, so when someone stumbles upon it in the future, they don't have to track down this PR discussion

after that, I think this is ready to go in :)